controllers/main.py

```diff
@@ -48,23 +67,23 @@ def get_shops_list(self, query=False,  debug=False, lat=False, lng=False, **kwar
                 'comment',
             ]
 
-	    isIds = request.params.get('ids','')
-	    tags = request.params.get('tags','')
-	    tags = tags.split(',') if tags else False
-            shop_ids = []
-            if not isIds:
-		#if not ids list passes get recordset of all shops ('public' flag in true)
-		#checks if linked warehouse record is not archieved ('active' flag is true)
-		#and filter by selected tags
-		domain = [('public', '=', True),('warehouse_ids.active','=',True)]
-		if tags:
-		    domain.append('|') if len(tags) > 1 else None
-		    for tag in tags:
-			tagId = request.env['res.partner.category'].sudo().search([('name','=',tag)]).id
-			domain.append(('category_id','in',tagId))
-                shop_ids = request.env['eyekraft.shop'].sudo().search(domain)
+        isIds = request.params.get('ids','')
+        tags = request.params.get('tags','')
+        tags = tags.split(',') if tags else False
+        shop_ids = []
+        if not isIds:
+            # if not ids list passes get recordset of all shops ('public' flag in true)
+            # checks if linked warehouse record is not archieved ('active' flag is true)
+            # and filter by selected tags
+            domain = [('public', '=', True),('warehouse_ids.active','=',True)]
+            if tags:
+                domain.append('|') if len(tags) > 1 else None
```

Comment:
It recommended to use expressions to work with domains
https://github.com/odoo/odoo/blob/13.0/odoo/osv/expression.py